stino-x/BUDGET-APP
# Models

This pull request introduces several changes to enhance the application's structure, models, and associations. Here's a summary of the changes:

Models:

Created User, Entity, and Group models to represent core entities in the application.
Established associations between these models to define their relationships.
Associations:

Defined associations between User, Entity, and Group models to reflect the application's data structure.
Implemented necessary migrations to set up the database schema with foreign keys.
Testing:

Introduced unit tests for each model using RSpec.
Ensured that the tests cover model validations and associations.
Tests for valid and invalid attributes, as well as associations, have been added to ensure the reliability of the models.
Configuration:

Configured the CanCanCan gem to manage user permissions and authorizations.
Set up the Ability class to define user roles and their corresponding abilities.
Usage of CanCanCan:

Integrated CanCanCan authorization checks in controllers to control access to specific actions based on user roles.
Miscellaneous:

Addressed issues related to foreign key naming in the database schema.
Ensured that the models and associations align with the intended application logic.
Testing Instructions:

Clone this branch and run bundle install.
Set up the database with rails db:create db:migrate.
Run RSpec tests with rspec to verify the correctness of the models and associations.
